From: Michael Froman <mfroman@mozilla.com>
Date: Tue, 1 Oct 2024 21:05:00 +0000
Subject: Bug 1921706 - use relative paths on deps in a few places to avoid
 future issues. r=ng,webrtc-reviewers DONTBUILD

Note: this produces no changes in the generated moz.build files.

Differential Revision: https://phabricator.services.mozilla.com/D224070
Mercurial Revision: https://hg.mozilla.org/mozilla-central/rev/bbec1b95ddb5e00967a0fb002f54f8f24336b904
---
 modules/audio_coding/BUILD.gn  |  2 +-
 modules/portal/BUILD.gn        |  8 ++++----
 modules/video_capture/BUILD.gn |  2 +-
 sdk/BUILD.gn                   | 14 ++++++++++++++
 test/BUILD.gn                  |  6 ++++++
 5 files changed, 26 insertions(+), 6 deletions(-)

diff --git a/modules/audio_coding/BUILD.gn b/modules/audio_coding/BUILD.gn
index 6fe5ff6a44..ab4854f8e3 100644
--- a/modules/audio_coding/BUILD.gn
+++ b/modules/audio_coding/BUILD.gn
@@ -540,7 +540,7 @@ rtc_library("webrtc_opus_wrapper") {
     deps += [ rtc_opus_dir ]
     public_configs = [ "//third_party/opus:opus_config" ]
   } else if (build_with_mozilla) {
-    public_configs = [ "//third_party/opus:opus_config" ]
+    public_configs = [ "../../third_party/opus:opus_config" ]
   }
 }
 
diff --git a/modules/portal/BUILD.gn b/modules/portal/BUILD.gn
index aff8609b54..901597ecde 100644
--- a/modules/portal/BUILD.gn
+++ b/modules/portal/BUILD.gn
@@ -137,10 +137,10 @@ if (!build_with_mozilla) {
       deps -= [ ":pipewire_stubs" ]
       defines -= [ "WEBRTC_DLOPEN_PIPEWIRE" ]
       public_deps = [
-        "//third_party/pipewire",
-        "//third_party/drm",
-        "//third_party/gbm",
-        "//third_party/libepoxy"
+        "../../third_party/pipewire",
+        "../../third_party/drm",
+        "../../third_party/gbm",
+        "../../third_party/libepoxy"
       ]
     }
   }
diff --git a/modules/video_capture/BUILD.gn b/modules/video_capture/BUILD.gn
index d9b3471572..60e191908b 100644
--- a/modules/video_capture/BUILD.gn
+++ b/modules/video_capture/BUILD.gn
@@ -110,7 +110,7 @@ if (!build_with_chromium || is_linux || is_chromeos) {
         ]
         if (build_with_mozilla) {
           configs -= [ "../portal:pipewire_base" ]
-          public_deps = [ "//third_party/pipewire" ]
+          public_deps = [ "../../third_party/pipewire" ]
         }
       }
     }
diff --git a/sdk/BUILD.gn b/sdk/BUILD.gn
index fce3ca62ff..d2a74716dd 100644
--- a/sdk/BUILD.gn
+++ b/sdk/BUILD.gn
@@ -588,6 +588,20 @@ if (is_ios || is_mac) {
         "CoreGraphics.framework",
         "CoreVideo.framework",
       ]
+      if (build_with_mozilla) {
+        deps -= [
+          "//api/video:video_frame",
+          "//api/video:video_rtp_headers",
+          "//common_video",
+          "//rtc_base:checks",
+        ]
+        deps += [
+          "../api/video:video_frame",
+          "../api/video:video_rtp_headers",
+          "../common_video",
+          "../rtc_base:checks",
+        ]
+      }
     }
 
     if (!build_with_mozilla) {
diff --git a/test/BUILD.gn b/test/BUILD.gn
index ac80805821..b4fd619f3d 100644
--- a/test/BUILD.gn
+++ b/test/BUILD.gn
@@ -483,6 +483,12 @@ rtc_source_set("test_support") {
     "//testing/gmock",
     "//testing/gtest",
   ]
+  if (build_with_mozilla) {
+    all_dependent_configs -= [
+      "//third_party/googletest:gmock_config",
+      "//third_party/googletest:gtest_config",
+    ]
+  }
 }
 
 rtc_library("fixed_fps_video_frame_writer_adapter") {
